applications/serial_lte_modem/src/http_c/slm_at_httpc.c

```diff
@@ -572,6 +575,11 @@ int handle_at_httpc_request(enum at_cmd_type cmd_type)
 
 	switch (cmd_type) {
 	case AT_CMD_TYPE_SET_COMMAND:
+		if (httpc_in_thread) {
```

Comment:
Is there a race condition where this check is passed for a second call, before the thread is created? That would overwrite the `data_buf` which holds the parameters for the first `http_client_req`.

Reply:
yes.
That is why I proposed that state check.

The bool or state have to be modified from the same thread that parses the AT command, so when ever it returns to parse a next command, the flag is set already.